payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncherViewModel.kt

```diff
@@ -83,12 +109,13 @@ internal class GooglePayLauncherViewModel(
 
     @VisibleForTesting
     internal fun createTransactionInfo(
-        stripeIntent: StripeIntent
+        stripeIntent: StripeIntent,
+        currencyCode: String
     ): GooglePayJsonFactory.TransactionInfo {
         return when (stripeIntent) {
             is PaymentIntent -> {
```

Comment:
By creating the PaymentIntent and SetupIntent objects it would be nice if you could use them interchangeably throughout the code.   I wonder if you could have both of them implement the same interface and for SetupIntents the SetupIntentArgs.totalAmount would return null, and total price status would be set based on the totalAmount == null

Reply:
We don't have the actual `SetupIntent` object in `SetupIntentArgs`, so I don't think that will work.

Reply:
right, and StripeSetupIntent doesn't have an amount member variable